src/regex_set.rs

```diff
@@ -8,8 +8,8 @@ use regex::RegexSet as RxSet;
 /// A dynamic set of regular expressions.
 #[derive(Debug)]
 pub struct RegexSet {
-    items: Vec<String>,
-    set: Option<RxSet>,
+    pub items: Vec<String>,
+    pub set: Option<RxSet>,
```

Comment:
By making these `pub`, we allow mutation, which we don't want. Instead, let's add getters that return immutable references: `&[String]` and `Option<&RxSet>` respectively.